configserver/src/main/java/com/yahoo/vespa/config/server/application/TenantApplications.java

```diff
@@ -185,7 +186,9 @@ public Lock lock(ApplicationId id) {
 
     private void childEvent(CuratorFramework ignored, PathChildrenCacheEvent event) {
         zkWatcherExecutor.execute(() -> {
-            ApplicationId applicationId = ApplicationId.fromSerializedForm(Path.fromString(event.getData().getPath()).getName());
+            ChildData data = event.getData();
+            if (data == null) return; // Node might have been deleted after we got event
```

Comment:
null is passed when connection is suspended, lost, or on (re)connected, or when cache is initialized(?)  So I think the correct thing to do is to duplicate the current statement to within each `case` in the below switch (and document getData may be null on some other event types)

Reply:
Right, thanks, will do as suggested